### ISSUE TITLE
How actually to use MacHyperVSupport without proper guide?

### ISSUE THREAD

**Issue by ANONIMNIQ** [acidanthera/bugtracker#2179]
Can someone use this? I don't really understand how to create macos vm for hyper-v. I can download installer.dmg ... then I need to create VHDX image from it, but how to do this from Windows? Also, how to attach OpenCore to the VM and where exctly to use MacHyperVSupport.kext? To put it in OpenCore and crete new cofig.plist maybe?

**Comment by pigworlds** [comment]
It is pretty simple to use MacHyperVSupport.kext under Hyper-V.

Prerequisites: Prepare the EFI folder and config.plist.
- You can start from https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Sample.plist as the base of your config.plist.
- Apply all necessary changes documented in https://github.com/acidanthera/MacHyperVSupport/blob/master/README.md. You don't need to apply all the changes unless you want to run a very old version of macOS.
- Add all the essential kexts including MacHyperVSupport.kext

Create the boot VHDX
- You can create a small 1GB VHDX, initialize as GUID partition table (GPT), and format as FAT32
- Place your EFI folder under the root. You should have directories EFI\Boot and EFI\OC. Technically you need to use EFI system partition, but Hyper-V recognizes regular FAT32 partition.

Prepare installer or recovery VHDX. (Only need to pick one)
- You can follow the doc https://github.com/acidanthera/MacHyperVSupport/blob/master/README.md about create and use the installer. This may need another Mac to do.
- You can also create a recovery VHDX. https://dortania.github.io/OpenCore-Install-Guide/installer-guide/winblows-install.html is your best friend to guide you, and replace the USB with the recovery VHDX you created.

Install macOS
- Create a Generation 2 VM
- Disable Secure Boot on the VM
- Attach the boot VHDX, installer/recovery VHDX created above
- Create an empty VHDX for the final macOS installation
- Boot the VM and select Recovery or Installer
- Follow the screen to install the operating system

You should have macOS running under Hyper-V now. Hope this helps.

Many thanks to the acidanthera team for making this happen.

**Comment by ANONIMNIQ** [comment]
Thank you! I'll tried it! btw is it possible to use some sort of gpu passthrough / GPU Partitioning to macos hyper-v vm?

**Comment by ANONIMNIQ** [comment]
@pigworlds can you share your EFI folder, I tried to make it but opencore can't show installer.

**Comment by pigworlds** [comment]
Have you put both recovery files under com.apple.recovery.boot folder?
You can take a look at my [config.plist](https://gist.github.com/pigworlds/870f6072889de4b8b25b05ab24c4236e#file-hyperv-opencore-config-plist). It adds MacHyperVSupport.kext and some cosmetic fixes.

**Comment by ANONIMNIQ** [comment]
@pigworlds I don't put recovery, I only convert installer.dmg to installer.vhdx and attached it to VM. With you config.plist it detects one hdd called NEW VOLUME in opencore and tried to boot from it but only shows black screen after that. I'm not sure if that's installer hdd or empty hdd for the vm.

**Comment by aguys123** [comment]
Sorry for OOT, @ANONIMNIQ.

@pigworlds I've successfully installed Mountain Lion in Hyper-V, thanks to your config (which I configured a little bit for older macos). Now, how to use hvfilecopyd, hvshutdownd, and hvtimesyncd daemons? Should I put it in /Library/LaunchDaemon?

**Comment by ANONIMNIQ** [comment]
I make a little progress but it stuck on boot:
https://ibb.co/tKwF3w0
https://ibb.co/4mxgRJb

**Comment by mahfoudfx** [comment]
Thanks @pigworlds  for your help.
Even if I carefully followed the documentation of OpenCore, I still unable to boot and struggling in `OpenShell.efi` boot.
I have a bunch of unclear points for which I didn't find no answer since several weeks, for example:

1. If I want to Install Monterey or Ventura, do I need only `MacHyperVSupportMonterey.kext` or to combine it with `MacHyperVSupport.kext`.
2. How to use the binaries: `hvfilecopyd`, `hvshutdownd `and `hvtimesyncd `mentioned in the beginning of the documentation of the MacHyperVSupport and that can be found under the Tools folder of the package.
3. Do we need to customize the `config.plist` according the used platform (CPU, SMBIOS) or as Virtual Environment it is not concerned.
4. The Installer Image creation is it based on the downloaded `BaseSystem.dmg` [Downloading macOS](https://dortania.github.io/OpenCore-Install-Guide/installer-guide/windows-install.html#downloading-macos) or another file? If I use the mentioned command `qemu-img convert -f raw -O vhdx Installer.dmg` Installer.vhdx I get an BaseSystem.img inside the `.vhdx` file, if I use it without the `-f raw`, it `.vhdx` containing macOS Base System partition but this will cause a failure to start the VM, maybe because the HFS file system is not recognized.
5. I see in your `config.plist` example the include of `HFSPlus.efi`, which is not included by default in OpenCore, is it mandatory?

If you can share the EFI folder, or write a proper guide for Hyper-V it will be very helpful.

**Comment by murkomarko** [comment]
does this still works?

**Comment by pidimensions** [comment]
I have done this successfully.
Hyper-V VM file: Hyper-V Virtual Machine: [Pre-Built VM with installer ready to go](Hyper-V Virtual Machine: https://drive.google.com/file/d/1hKnl8DnO0yOhqT4lk7TXaabHnW8oLGHZ/view?usp=sharingl)
EFI Folder: [](https://drive.google.com/file/d/1k282wWfil8dy4VnsSNotIrYVOLMZpYCx/view?usp=sharing)

**Comment by UsedDiscord** [comment]
Are you really sure it works? Because it doesn't. @pidimensions Once past the openCore boot, we have a forbidden sign after selecting the DMG...

**Comment by pidimensions** [comment]
Hm, it worked for me. I'll take a look and try it again

**Comment by zeanzhou** [comment]
I got the forbidden sign as well, but it turns out that I allocate too few memory to MacOS..

* 4096MB - Doesn't work
* 8192MB - It works, installation continues.

More background: 
* I used the EFI folder provided by @pidimensions , and combined it with the dmg by following [this guide ](https://dortania.github.io/OpenCore-Install-Guide/installer-guide/windows-install.html#disk-management-method) (Putting both folders together into the vhdx)
* The Hyper-V VM from @pidimensions works for me (I didn't continue with the installation, but I can see the language selection step. Note that since I don't have 16000MB memory installed, the configuration in this VM failed for me, actually I had to tweak this and some other settings a bit to make it work. But this finding helped me realize my issue :D )

Hope it helps.

**Comment by UsedDiscord** [comment]
> I got the forbidden sign as well, but it turns out that I allocate too few memory to MacOS..

> 4096MB - Doesn't work
> 8192MB - It works, installation continues.

Mmh, I'm interested, I've actually also put 4GB of RAM, I'll test with 8, thanks for the help! @zeanzhou

**Comment by cmd430** [comment]
> How much RAM is allocated to the VM @cmd430 ? And did you use my EFI/prebuilt VM

16GB, using your prebuilt VM, I am however using an AMD CPU in this PC, so that could be the issue

**Comment by cmd430** [comment]
ah i brain farted, never actually checked in Disk Manager after attempting to mount....

I did notice however that OpenCore claims to support AMD CPUs
> CPU Support [...] Ryzen (17h) CPUs 
I am using 5600x (zen 3/19h) so maybe this will be useful information, maybe its not possible but hopefully its something relatively easy to get working

**Comment by pidimensions** [comment]
Okay, what model is your CPU and how many cores does it have? I can held with kernel patching

**Comment by cmd430** [comment]
> Okay, what model is your CPU and how many cores does it have? I can help with kernel patching which is needed Edit: the amount of cores that the VM will have allocated, must be 4, 6, 8, 12, 16, 24, 32

Fixed my last message formatting, I have a Zen3 5600X, it is a 6 core CPU (12 Threads)

**Comment by pidimensions** [comment]
Okay do you plan to allocate all of the cores to the VM?

**Comment by cmd430** [comment]
No was planning on allocating 4

**Comment by pidimensions** [comment]
Ok I'll change the config.plist based on that and post another version with google drive link
https://dortania.github.io/OpenCore-Install-Guide/AMD/fx.html#deviceproperties

**Comment by pidimensions** [comment]
Alright yeah that's probably an AMD CPU issue, I have attached here the fixed VM for a 4 core AMD CPU. It should work
[4-Core AMD VM](https://drive.google.com/file/d/1f2GVmWkDCdqdV0PB1GEeh9eZM7E4PMSp/view?usp=sharing)

**Comment by aguys123** [comment]
According to [Goldfish64](https://github.com/Goldfish64) : 
> AMD is currently completely unsupported and requires additional patches beyond the standard AMD macOS patches.

https://github.com/acidanthera/bugtracker/issues/2120#issuecomment-1250372560

**Comment by pidimensions** [comment]
Well it's worth it for @cmd430 to at least give it a try

**Comment by cmd430** [comment]
Confirmed still unable to load the installer, was worth a shot

**Comment by pidimensions** [comment]
Ah damn that's unfortunate

**Comment by TaylorTWBrown** [comment]
> I have done this successfully. Attached are the files for anyone wanting to do this [Hyper-V VM](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing) [EFI Folder](https://drive.google.com/file/d/1k282wWfil8dy4VnsSNotIrYVOLMZpYCx/view?usp=sharing)
> 
> Many thanks to @pigworlds for sharing the config.plist

Thank you for sharing this! I imported the VM and unfortunately it just hangs on the Apple logo.

**Comment by pidimensions** [comment]
There have been many issues with RAM allocation, how much did you give to the VM? 8 GB and 16 GB are good, anything else might cause boot issues

**Comment by TaylorTWBrown** [comment]
I used 16000 MB, the same that's set in the VM. 8000, 8192, and 16384 seem to yield the same result.

I've also tried 4 and 8 cores. Could it have to do with Windows 11 and the [Hypervisor scheduler](https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/manage/manage-hyper-v-scheduler-types), or the performance/efficiency cores of the 12th gen CPU.

**Comment by pidimensions** [comment]
Ah damn I don't know then, okay yeah maybe the other computer will be better

**Comment by TaylorTWBrown** [comment]
I just tried it on a server I have running. Windows Server 2022, Xeon E5 2680 v4.

I get the same issue. The one thing that my desktop and server have in common, is that they're both running the latest version of Windows. If you're running Windows 10 or Server 2016, maybe something is substantially different.

I'm curious: @pidimensions If you download and import the VM that you uploaded, does it 'just work' for you? What version of Windows are you running, and what's your CPU?

**Comment by pidimensions** [comment]
I'm running Windows 10 with i7-6700k

**Comment by pidimensions** [comment]
And it does work as shown here
[video](https://www.youtube.com/watch?v=Ob8KW2YwXk4)

**Comment by TaylorTWBrown** [comment]
There might be some bigger hypervisor differences with Windows 10. I'll report back soon, hopefully.

**Comment by pidimensions** [comment]
Okay, I'd be interested to know if its different on different versions of Windows

**Comment by aguys123** [comment]
> I get the same issue. The one thing that my desktop and server have in common, is that they're both running the latest version of Windows. If you're running Windows 10 or Server 2016, maybe something is substantially different.


I'm on Windows 11 and successfully installed Mountain Lion and High Sierra with only 4GB of RAM allocated (on Acer laptop, i7-8550U proc, 16GB RAM). Maybe you can try spoofing the CPU.

**Comment by pigworlds** [comment]
> > I have done this successfully. Attached are the files for anyone wanting to do this [Hyper-V VM](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing) [EFI Folder](https://drive.google.com/file/d/1k282wWfil8dy4VnsSNotIrYVOLMZpYCx/view?usp=sharing)
> > Many thanks to @pigworlds for sharing the config.plist
> 
> Thank you for sharing this! I imported the VM and unfortunately it just hangs on the Apple logo.
> 
> I'm running an Intel i9-12900K.
> 
> ![image](https://user-images.githubusercontent.com/20453802/242715753-4971eabe-5242-445a-ac11-562c4d40d7b0.png)

I got success with some newer Intel CPU using CPUID spoofing. 
You can give a try to modify this section in the config.plist
```xml
		<key>Emulate</key>
		<dict>
			<key>Cpuid1Data</key>
			<data>6gYJAAAAAAAAAAAAAAAAAA==</data>
			<key>Cpuid1Mask</key>
			<data>/////wAAAAAAAAAAAAAAAA==</data>
```

**Comment by TaylorTWBrown** [comment]
> ```
> 		<key>Emulate</key>
> 		<dict>
> 			<key>Cpuid1Data</key>
> 			<data>6gYJAAAAAAAAAAAAAAAAAA==</data>
> 			<key>Cpuid1Mask</key>
> 			<data>/////wAAAAAAAAAAAAAAAA==</data>
> ```

Thank you! That's all I needed to get booted. I'll finish my install tomorrow, and happily share my progress and any relevant files.

**Comment by aguys123** [comment]
> Interesting, should I add that change into the Google Drive link VM? Or is it CPU-specific?

I think it's host CPU related since I cannot boot Mountain Lion VM without spoofing CPU to Ivy Bridge and I can boot High Sierra without spoofing (Kabylake supported by HS).

**Comment by zeanzhou** [comment]
Hello all, I am trying to activate iMessage on this VM, but the iMessage App just logged me out a couple of seconds after logging in. When I run the iMessage App from terminal, I can see the customer code error being printed out in the terminal, the message is the same as the one in this guide https://dortania.github.io/OpenCore-Post-Install/universal/iservices.html . 

Before calling Apple, I was still trying to troubleshoot this error, and I did go through every single step in the above guide. However, when I followed `What if I don't have En0 at all?!?` step, even after applying `NullEthernet.kext`, I still got an empty list in Hackintool as below.
![hackintool](https://github.com/acidanthera/bugtracker/assets/17812593/17a549fc-b78d-4f11-bf86-0138f85436c2)

Note that my network is working. I can see `en0` and a fake `en1` after applying `NullEthernet.kext` via `ifconfig`. 

And when I switched to `PCIe` Tab in Hackintool, it only shows 1 device as below. But the same screenshot in the guide contains a long list of PCI devices, which makes me feel something is wrong with my hyper-v configuration.
![hackintool2](https://github.com/acidanthera/bugtracker/assets/17812593/078d367a-c294-40ac-8f58-e4ca65ac587c)


Background:
* Boot: EFI folder from @pidimensions (Thanks!!)
* Serial Number is generated using macserial
* Product Code: MacBookPro14,1
  * Kernel crashed at boot using iMacPro1,1
  * Kernel crashed at boot using MacPro7,1 (I am confused tho, someone [is able to boot](https://github.com/GeekyCoder7/OpenCore-EFI-Lenovo-Y50-70) using this code in a real device the same as mine?)
  * Macmini8,1 is bootable but iMessage doesn't work for me either
* Network is working without issue.
* AppleId logging is working without issue.
* Guest: Ventura, 2 virtual processors, 8GB RAM
* Host: Windows 10 Version 22H2, i7-4710HQ, 16GB RAM


**Could someone help to confirm this is the right behavior? Are you also seeing an empty list there using Hackintool, and 1 device only in `PCIe` Tab? Are you able to make iMessage work with Hyper-V?**

Not sure whether this is the right place to ask this question, let me know if there is a better place!

**Comment by pidimensions** [comment]
Yeah I am experiencing the same thing with no Ethernet Peripherals in the System > Peripherals but an ethernet PCIe device

**Comment by TaylorTWBrown** [comment]
@zeanzhou from the screenshot, I see you have Bluetooth in the VM. How did you passthrough the Bluetooth adapter, and what hardware are you using?

**Comment by pidimensions** [comment]
@TaylorTWBrown he doesn't have Bluetooth. That's shown for me as well. It's likely a dummy that OpenCore creates

**Comment by pidimensions** [comment]
![image](https://github.com/acidanthera/bugtracker/assets/125328243/a0dfbf98-6f27-4f61-8a9c-ce64d6a090ea)

**Comment by TaylorTWBrown** [comment]
> @TaylorTWBrown he doesn't have Bluetooth. That's shown for me as well. It's likely a dummy that OpenCore creates

Interesting. For me, one motivation for macOS on Hyper-v is to passthrough Intel AX201 or CSR4.0 Bluetooth (using USBIP or something). No luck yet - I get it to turn on with the adapters connected, but I can't actually scan/find devices. Likely something to do with loading firmware at boot time, or sharing of a CNVio device (?). 

@zeanzhou I tried signing in to messages, and it also crashes. You may want to open a new issue to get more visibility.

**Comment by aguys123** [comment]
![Screenshot 2023-06-07 212204](https://github.com/acidanthera/bugtracker/assets/83843623/d8ed15c9-07da-4861-85e9-53769512b3fc)

Slow but smooth install...

**Comment by pidimensions** [comment]
Damn nice

**Comment by zjy2931** [comment]
How much time did you spend on installing this? It has taken me an hour and the Apple loading bar is still not done.

**Comment by zjy2931** [comment]
> ![Screenshot 2023-06-07 212204](https://user-images.githubusercontent.com/83843623/244080393-d8ed15c9-07da-4861-85e9-53769512b3fc.png)
> 
> Slow but smooth install...

How much time did you spend on installing this? It has taken me an hour and the Apple loading bar is still not done.

**Comment by aguys123** [comment]
> How much time did you spend on installing this? It has taken me an hour and the Apple loading bar is still not done.

An hour and a half, I guess. VM files location in HDD though.

**Comment by ANONIMNIQ** [comment]
What is the performance? Also, can you use it for video editing and rendering without GPU acceleration?

**Comment by healenrens** [comment]
Hi, Excuse me.I would like to know if anyone has successfully run on an AMD CPU now? I tried to modify according to the guidelines provided by OpenCore. Unfortunately, it seems different from running on a virtual machine. I am using 5700G. If someone has successfully run on an AMD CPU, could you please provide the plist file and attach your CPU/motherboard model as a reference? Thank you very much.

**Comment by DrakkLord** [comment]
Hello there, I Just wanted to give help to others who may stumble upon this that I was able to get it working. Thanks to @pigworlds  suggestions I was able to get it fully working. My case was interesting because I have an installed Monterey on the same hardware I run Windows on so I knew beforehand that the PC can run the OS natively so I has to work somewhat in hyper-v.

```
CPU: Intel i7-12700KF
Windows build: 22621.1848 
```

The OpenCore configuration that pigworlds provided originally only needed one other addition that he mentioned in a later comment, the CPUID mod:

		<key>Emulate</key>
		<dict>
			<key>Cpuid1Data</key>
			<data>6gYJAAAAAAAAAAAAAAAAAA==</data>
			<key>Cpuid1Mask</key>
			<data>/////wAAAAAAAAAAAAAAAA==</data>

For me this was essential otherwise the kernel got stuck on boot. I still have weird stuck boots but restarting almost always solves it.

My target was to install Ventura and Sonoma as well as Monterey, all succeded. I was able to download the Ventura installer on Monterey and clean install Ventura from that, for Sonoma I had to mess around a little bit, I ended up moving the Sonoma installer to a separate vhdx and install from that. I made a recovery image as how the OpenCore documentation said and used a Monterey recovery to install Ventura, Monterey and a Ventura recovery to install Sonoma.

------
![ventura-installed](https://github.com/acidanthera/bugtracker/assets/7610117/6ce0245a-376d-4c8c-9e32-4e1a43fc01fc)

------

![sonoma-installed](https://github.com/acidanthera/bugtracker/assets/7610117/bf06c931-8add-45e5-888d-f7b66cfeb2c8)

**Comment by infortek1991** [comment]
Hallo, Can I please for GOOD Instruction to do this ? Please form help and understand ? I don't understand how to create this all ? Please for Help.

**Comment by pidimensions** [comment]
Hello infortek1991. I sympathize with you struggles, however, I have managed to get this working on my Windows 10 system with an i7-6700k using this VM file here
[Hyper-V VM File](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing)

**Comment by DovydasTEDS** [comment]
Has anyone managed to get GPU passthrough working?

**Comment by DrakkLord** [comment]
> Has anyone managed to get GPU passthrough working?

Yes I was able to, I don't remember the exaxt steps I followed but it only works with windows guests and only with Direct 3D. It was a somewhat complicated process, enabling on the guest then copying driver files to the guest. So it worked but for me it was not good because I wanted to have vulkan supported on the passtrough GPU and non-windows OS as well. So I tried it and just didnt bother using it because of this. I think rhis was the repo/guide : https://github.com/jamesstringerparsec/Easy-GPU-PV

**Comment by DovydasTEDS** [comment]
@DrakkLord 

I have used the repo before, with a Windows guest and it ran alright. It had some lag, lower performance and, like you mentioned, only D3D was passed though unfortunately. I hope Microsoft can somehow improve GPU passthrough, which might also enable us to use GPU passthrough on a macOS VM.

**Comment by infortek1991** [comment]
> Hello infortek1991. I sympathize with you struggles, however, I have managed to get this working on my Windows 10 system with an i7-6700k using this VM file here [Hyper-V VM File](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing)

Thank You for Your reply and help. I will try to use what You shared. Thank You very much again for Your help. :)

**Comment by pidimensions** [comment]
> > Hello infortek1991. I sympathize with you struggles, however, I have managed to get this working on my Windows 10 system with an i7-6700k using this VM file here [Hyper-V VM File](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing)
> 
> Thank You for Your reply and help. I will try to use what You shared. Thank You very much again for Your help. :)

You're very welcome. Hope you are successful!

**Comment by vnxnam** [comment]
Hello, I installed well-done. But Hyper-V Graphics only 3MB and resolution screen 1024x786 (default) . How can I increase Graphics more to up 128MB and to use UIScale HiDPI mode?
Thank you

**Comment by rafail-r** [comment]
> > > Hello infortek1991. I sympathize with you struggles, however, I have managed to get this working on my Windows 10 system with an i7-6700k using this VM file here [Hyper-V VM File](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing)
> > 
> > 
> > Thank You for Your reply and help. I will try to use what You shared. Thank You very much again for Your help. :)
> 
> You're very welcome. Hope you are successful!

thank you for your video. I managed to finish installation but I m getting this error. Do you have any advice for me where to look for a solution? 
![image](https://github.com/acidanthera/bugtracker/assets/18389355/e172ed4a-0465-46e3-b8c0-de5fa49a5dcc)

**Comment by rafail-r** [comment]
> > > > Hello infortek1991. I sympathize with you struggles, however, I have managed to get this working on my Windows 10 system with an i7-6700k using this VM file here [Hyper-V VM File](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view?usp=sharing)
> > > 
> > > 
> > > Thank You for Your reply and help. I will try to use what You shared. Thank You very much again for Your help. :)
> > 
> > 
> > You're very welcome. Hope you are successful!
> 
> thank you for your video. I managed to finish installation but I m getting this error. Do you have any advice for me where to look for a solution? ![image](https://user-images.githubusercontent.com/18389355/257056993-e172ed4a-0465-46e3-b8c0-de5fa49a5dcc.png)

I managed to skip various errors by saving (through hyper-v) and then starting again when the apple logo with the load bar is showing. After resuming from saved state the bar seemed to load way faster for some reason

**Comment by auspicious946** [comment]
I'm using an AMD CPU(Ryzen  5 4600H). The OS doesn't seem to work with me. I use the EFI file provided by @pidimensions. That doesn't work on my PC. My VM is struck on the Apple logo 
![image](https://github.com/acidanthera/bugtracker/assets/78522158/c5b34c83-fa43-4164-a7e9-9582dc736030)

**Comment by pidimensions** [comment]
> I'm using an AMD CPU(Ryzen 5 4600H). The OS doesn't seem to work with me. I use the EFI file provided by @pidimensions. That doesn't work on my PC. My VM is struck on the Apple logo
> ![image](https://user-images.githubusercontent.com/78522158/260316596-c5b34c83-fa43-4164-a7e9-9582dc736030.png)

Unfortunately AMD CPUs seem to not work in hyper-v

**Comment by mio-19** [comment]
> mine gets the the apple logo but never progresses past that (no progress bar, just black screen with apple logo)

I got the same result on 12gen intel with 8192MB RAM allocated

**Comment by Pnc8922676198** [comment]
I can't use this this file [](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view),my i use i7 and win11,who can help me solve this problem? I'm not very familiar with this.

**Comment by lucasfcnunes** [comment]
In my case:
- I used @pidimensions [EFI - CPU spoof for newer CPUs.zip](https://drive.google.com/file/d/11sfgMR5M0vU-MDdi4KNXNy5VgCEnNhEt/view) and [MacOS Ventura Hyper-V VM - Intel CPU.zip](https://drive.google.com/file/d/1XuzQw70lhWOZ6IojCA1qZuunmV6DjJ2t/view).
- If it takes more than 2 min (gets stuck) reset/restart the VM. I didn't change anything and it worked.
- 16000 MiB RAM and 8 CPU were kept + I had to add a network with internet (Default network) because it complained that it needed internet access to continue with the installation.
- I actually:
  - Mounted the original Install.vhdx
  - Gave it a drive letter on Windows' Disk Management
  - Deleted /EFI and add /EFI/EFI folder with EFI - CPU spoof for newer CPUs.zip content (drag and drop of /EFI/EFI from in winrar)
  - Made it offline in Disk Management to be able to use in the VM

PS:
  - If you get support.apple.com/mac/startup (https://support.apple.com/en-us/101666) error, reset/restart the VM and keep the mouse in the VM screen
  - Hyper-V Disk.vhdx is just an empty vhdx file
  - I will use Install.vhdx directly after I move and rename to macos_boot.vhdx or something else
  - https://www.youtube.com/watch?v=Ob8KW2YwXk4 is a good reference

![image](https://github.com/acidanthera/bugtracker/assets/29926675/69052e30-a3ed-469b-9971-c2426e8e8580)

---
MOBO: HUANANZHI X99 F8D PLUS
CPU: 2x/DUAL Xeon E5 2696v4 (2016, 14 nm, Haswell like)
RAM: 128 GiB (4x 32 GiB)
OS Name Microsoft Windows 10 Pro
OS Version: 10.0.19045 N/A Build 19045

**Comment by pidimensions** [comment]
Great, I am very glad to hear this success! Thanks for contributing

**Comment by lucasfcnunes** [comment]
> Sorry for OOT, @ANONIMNIQ.
> 
> @pigworlds I've successfully installed Mountain Lion in Hyper-V, thanks to your config (which I configured a little bit for older macos). Now, how to use hvfilecopyd, hvshutdownd, and hvtimesyncd daemons? Should I put it in /Library/LaunchDaemon?

Now I have the same question.

**Comment by nmocruz** [comment]
I used steps from @lucasfcnunes, I have the same  MB, and the Xeons are different and a bit less memory, but the main difference is Windows 11 instead of Windows 10.

In my case, the install get stuck on the Apple logo.

I have tried a clean install following the README but no luck I end up stuck on boot

I am attaching my logs here to see if someone points the problem on my setup

[opencore-2023-08-31-151349.txt](https://github.com/acidanthera/bugtracker/files/12487294/opencore-2023-08-31-151349.txt)

**Comment by nmocruz** [comment]
ok maybe is my graphics card, is a 550 RX a lexa

**Comment by pidimensions** [comment]
Damn YouTube took down my video :(

**Comment by lucasfcnunes** [comment]
> Damn YouTube took down my video :(

@pidimensions was it Apple?

**Comment by pidimensions** [comment]
> > Damn YouTube took down my video :(
> 
> @pidimensions was it Apple?
Idk it just just that "our team thinks it violates the content policy" for harmful or dangerous content

**Comment by balopez83** [comment]
Hi All, I have actually created a repo to make macOS on Hyper-V a little easier. Not only do I have additional instructions but I have created the VHDX drive needed to boot on most Intel CPU's. Currently it should support all macOS native CPU's as well as 11th and newer CPU's.

Of course it goes without saying that GPU passthrough is not possible, and without graphics acceleration you should expect sluggishness and weird graphics from time to time on the newer macOS versions. Within my guide I have also added some options to speed up macOS to make it far less sluggish even without graphics acceleration. 

You can find my repo [Here](https://github.com/balopez83/macOS_On_Hyper-V) which is updated when needed to ensure I am using the latest relevant hyper-v releases from acidanthera.

**Comment by nmocruz** [comment]
Would be nice to have GPU passthrough support, I think without that the mac
performance it will make it usable just for certain pontual tasks.
But I will test it later. I am having issues to make it work on my dual
xeon setup. I get stuck on boot even with a ready image.
